uv_copy_and_paste_uv/cpuv_transfer_uv_operation.py

```diff
@@ -87,47 +97,57 @@ def execute(self, context):
 
         uv_layer = bm.loops.layers.uv.verify()  # another approach
 
-        all_sorted_faces = main_parse(self, active_obj, bm, uv_layer)
-        if all_sorted_faces:
-            # check amount of copied/pasted faces
-            if len(all_sorted_faces) != len(props.topology_copied):
-                self.report({'WARNING'}, "Mesh has different amount of faces!!")
-                return {'CANCELLED'}
+        # get selection history
+        all_sel_faces = [e for e in bm.select_history if isinstance(e, bmesh.types.BMFace) and e.select]
+        print(all_sel_faces)
+        if len(all_sel_faces) % 2 != 0:
+            self.report({'WARNING'}, "Every part should select two faces!!")
+            return {'CANCELLED'}
+
+        # parse selection history
+        for i in range(len(all_sel_faces)):
+            if i > 0 and i % 2 != 0:
+                print(i)
+                sel_faces = [all_sel_faces[i-1], all_sel_faces[i]]
+                active_face = all_sel_faces[i]
+
+                # parse all faces according to selection history
+                all_sorted_faces = main_parse(self, active_obj, bm, uv_layer, sel_faces, active_face)
 
-            for i, face_data in enumerate(all_sorted_faces.values()):
-                copied_data = props.topology_copied[i]
+                if all_sorted_faces:
+                    # check amount of copied/pasted faces
+                    if len(all_sorted_faces) != len(props.topology_copied):
+                        self.report({'WARNING'}, "Mesh has different amount of faces!!")
+                        return {'CANCELLED'}
 
-                # check amount of copied/pasted verts
-                if len(copied_data[0]) != len(face_data[2]):
-                    bpy.ops.mesh.select_all(action='DESELECT')
-                    list(all_sorted_faces.keys())[i].select = True  # select problematic face
+                    for i, face_data in enumerate(all_sorted_faces.values()):
+                        copied_data = props.topology_copied[i]
 
-                    self.report({'WARNING'}, "Face have different amount of verts!!")
-                    return {'CANCELLED'}
+                        # check amount of copied/pasted verts
+                        if len(copied_data[0]) != len(face_data[2]):
+                            bpy.ops.mesh.select_all(action='DESELECT')
+                            list(all_sorted_faces.keys())[i].select = True  # select problematic face
```

Comment:
If this line comes, grown faces is selected after operation?

IMO, selected face is revert to before operation.